zerver/views/message_fetch.py

```diff
@@ -792,7 +839,7 @@ def parse_anchor_value(anchor_val: Optional[str],
         raise JsonableError(_("Invalid anchor"))
 
 @has_request_variables
-def get_messages_backend(request: HttpRequest, user_profile: UserProfile,
+def get_messages_backend(request: HttpRequest, maybe_user_profile: UserProfile,
```

Comment:
```suggestion
def get_messages_backend(request: HttpRequest, maybe_user_profile: Union[AnonymousUser, UserProfile],
```

unless I'm misunderstanding what kind of object is being passed in here.

Reply:
In my understanding, if the user is not authenticated, `maybe_user_profile` is always None.

Reply:
Mateusz is correct, it will be an AnonymousUser.  See https://docs.djangoproject.com/en/3.0/ref/contrib/auth/#django.contrib.auth.models.AnonymousUser; basically that's what Django uses as `request.user` in unauthenticated requests.